wandb/sdk/lib/gitlib.py

```diff
@@ -104,13 +127,13 @@ def last_commit(self):
             return None
 
     @property
-    def branch(self) -> Optional[str]:
+    def branch(self) -> Any:
         if not self.repo:
             return None
         return self.repo.head.ref.name
```

Comment:
should we catch the assertion error as well?

Reply:
we do similar in has_submodule_diff..  so i think without better tests in this codepath id prefer minimal changes.